### PR TITLE
fix: updated regclient get requests to handle query params

### PR DESCRIPTION
**Pull request by osterbit**

### CODE REVIEW

File: regclient/search.go [hunk 1]
```diff
@@ -49,21 +49,51 @@ func (c Client) Search(p *SearchParams) ([]*registry.Result, error) {
...
+	req.URL.RawQuery = q.Encode()
+	return req, nil
+}
 
-	req, err := http.NewRequest(method, fmt.Sprintf("%s/search", c.cfg.Location), bytes.NewReader(data))
+func (c Client) doJSONSearchReq(method string, s *registry.SearchParams) (results []*registry.Result, err error) {
+	if c.cfg.Location == "" {
+		return nil, ErrNoRegistry
+	}
+	req := &http.Request{}
```
Comment (b5) [inline regclient/search.go]: this should be: 
```go
var req *http.Request
```
This code is currently allocating an complete `http.Request` struct, allocating a pointer, and then never using the struct, as all cases in the following `switch` statement re-assigning the value of `req`.

File: regclient/search.go [hunk 2]
```diff
@@ -85,6 +115,6 @@ func (c Client) doJSONSearchReq(method string, s *registry.SearchParams) ([]*reg
 	if res.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("error %d: %s", res.StatusCode, env.Meta.Error)
 	}
-
-	return env.Data, nil
+	results = env.Data
```
Comment (b5) [inline regclient/search.go]: this could be refactored to
```go
return env.Data, nil
```

File: regclient/search.go [hunk 3]
```diff
@@ -49,21 +49,51 @@ func (c Client) Search(p *SearchParams) ([]*registry.Result, error) {
...
+	return req, nil
+}
 
-	data, err := json.Marshal(s)
+func (c Client) prepGetReq(s *registry.SearchParams) (*http.Request, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/search", c.cfg.Location), nil)
 	if err != nil {
 		return nil, err
 	}
+	q := req.URL.Query()
```
Comment (b5) [inline regclient/search.go]: 👌

**Comment by osterbit** [comment]
superseded by #9

**Pull request closed without merging**